Extending/FileSystemProviders/index.md

```diff
@@ -1,48 +1,159 @@
 ---
-versionFrom: 7.0.0
-needsV8Update: "true"
+versionFrom: 8.0.0
 ---
 
 # Custom file systems (IFileSystem)
 
-By default, Umbraco uses an instance of `PhysicalFileSystem` to handle the media archive. The configuration for this can be found in `/config/FileSystemProviders.config`.
+By default, Umbraco uses an instance of `PhysicalFileSystem` to handle the storage location of the media archive (~/media).
 
-```xml
-<Provider alias="media" type="Umbraco.Core.IO.PhysicalFileSystem, Umbraco.Core">
-    <Parameters>
-        <add key="virtualRoot" value="~/media/" />
-    </Parameters>
-</Provider>
-```  
+This can be configured during composition:
+
+```csharp
+using Umbraco.Core.Composing;
+using Umbraco.Core;
+using Umbraco.Core.IO;
+
+namespace Umbraco8Examples.Composition
+{
+    public class SetMediaFileSystemComposer : IUserComposer
+    {
+        public void Compose(Umbraco.Core.Composing.Composition composition)
+        {
+            composition.SetMediaFileSystem(() => new PhysicalFileSystem("~/custommediafolder"));
+        }
+    }
+}
+```
+The PhysicalFileSystem accepts the application path to the 'virtual Root' to location the Media folder the ~ is important!
+
+Alternatively you can combine:
+a **RootPath** - the rooted, filesystem path, using directory separator chars, NOT ending with a separator // eg "c:" or "c:\path\to\site" or "\\server\path"
+and **RootUrl** - the relative url, using url separator chars, NOT ending with a separator // eg "" or "/Views" or "/Media" or "/<vpath>/Media" in case of a virtual path
+
```

Comment:
To the above:
As I understand it, you can **combine** `RootPath` and `RootUrl`, right?
Would it make sense to add an example maybe? 🤔

Reply:
@sofietoft added in latest push...